org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/WorkspaceManager.java

```diff
@@ -466,9 +467,12 @@ public <T> T doRead(URI uri, Function2<? super Document, ? super XtextResource,
 		URI resourceURI = uri.trimFragment();
 		ProjectManager projectMnr = getProjectManager(resourceURI);
 		if (projectMnr != null) {
-			XtextResource resource = (XtextResource) projectMnr.getResource(resourceURI);
-			Document doc = getDocument(resource);
-			return Tuples.pair(doc, resource);
+			Resource resource = projectMnr.getResource(resourceURI);
+			if (resource instanceof XtextResource) {
```

Comment:
the else case is in the case the fall-through and returning null. The caller (the method right above) handles that by calling work.apply with null as document/resource.

Reply:
thats clear. i am just wonder if and how we dont support non xtext resources at all properly

Reply:
@cdietrich , Okay, I understand now. I also do not know the answer. Later this year we will take a look because we have in scope dealing with XmlResource. For the UI plugins and the UI builder, we did manage to make it work to some extent and we should do the same for the IDE plugins and the LS builder.
For the time being this PR does not answer the question but I think it is better not to crash if we can handle it more gracefully. So I would be happy if we could merge the small improvement.